src/main/java/org/terasology/lightandshadowresources/logic/domino/DominoSystem.java

```diff
@@ -0,0 +1,230 @@
+        if (targetBlockComp == null) {
+            event.consume();
+            return;
+        }
+
+        Vector3f horizDir =
+                new Vector3f(event.getDirection())
+                        .setComponent(1, 0); // set y dimension to 0
+        Side facingDir = Side.inDirection(horizDir);
+        if (!facingDir.isHorizontal()) {
+            event.consume();
+            return;
+        }
+
+        Vector3ic blockPos = targetBlockComp.getPosition();
+        Side offsetDir = facingDir.reverse();
+        Vector3i primePos = blockPos.add(offsetDir.direction(), new Vector3i());
+        Block primeBlock = worldProvider.getBlock(primePos);
+        if (!primeBlock.isReplacementAllowed()) {
+            event.consume();
+            return;
+        }
+
+        Block belowBlock = worldProvider.getBlock(primePos.x, primePos.y - 1, primePos.z);
+        Block aboveBlock = worldProvider.getBlock(primePos.x, primePos.y + 1, primePos.z);
+
+        // Determine top and bottom blocks
+        Vector3i bottomBlockPos = new Vector3i();
+        Vector3i topBlockPos = new Vector3i();
+        if (belowBlock.isReplacementAllowed()) {
+            bottomBlockPos.set(primePos.x, primePos.y - 1, primePos.z);
+            topBlockPos.set(primePos);
+        } else if (aboveBlock.isReplacementAllowed()) {
+            bottomBlockPos.set(primePos);
+            topBlockPos.set(primePos.x, primePos.y + 1, primePos.z);
+        } else {
+            event.consume();
+            return;
+        }
+        Side attachSide = determineAttachSide(facingDir, offsetDir, bottomBlockPos, topBlockPos);
+        if (attachSide == null) {
+            event.consume();
+            return;
+        }
+
+        Side risenSide = facingDir.reverse();
+
+        Block newBottomBlock = domino.bottomBlockFamily.getBlockForPlacement(new BlockPlacementData(bottomBlockPos,
+                risenSide, TOP));
+        Block newTopBlock = domino.topBlockFamily.getBlockForPlacement(new BlockPlacementData(bottomBlockPos,
+                risenSide, TOP));
+
+        Map<org.joml.Vector3i, Block> blockMap = new HashMap<>();
+        blockMap.put(bottomBlockPos, newBottomBlock);
+        blockMap.put(topBlockPos, newTopBlock);
+        PlaceBlocks blockEvent = new PlaceBlocks(blockMap, event.getInstigator());
+        worldProvider.getWorldEntity().send(blockEvent);
+
+        if (!blockEvent.isConsumed()) {
+            EntityRef newDomino = entityManager.create(domino.dominoRegionPrefab);
+            entity.removeComponent(MeshComponent.class);
+
+            newDomino.addComponent(new BlockRegionComponent(new BlockRegion(bottomBlockPos).union(topBlockPos)));
+
+            Vector3fc dominoCenter = new Vector3f(bottomBlockPos).add(0, 0.5f, 0);
+            newDomino.addComponent(new LocationComponent(dominoCenter));
+
+            DominoComponent newDominoComp = new Domino.getComponent(DominoComponent.class);
+            newDominoComp.risenSide = risenSide;
```

Comment:
Your `DominoComponent` does not have a field called `risenSide`. Did you forget to commit/push any local changes you made to the `DominoComponent` class? (same for `fallSide` and `isFallen`)

Reply:
@jdrueckert Like I said I copied code from Door's component so I forgot to rename `openSide` and `closedSide` in `door.component`, my bad, I overlooked that part of the code. I will fix that right away.